proposals/XXXX-limit-textfield-length-according-to-hmi-capabilities.md

```diff
@@ -0,0 +1,56 @@
+# Limit TextField Length According to HMI Capabilities
+
+* Proposal: [SDL-XXXX](XXXX-limit-textfield-length-according-to-hmi-capabilities.md)
+* Author: [Collin McQueen](https://github.com/iCollin)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+This proposal defines how Core should limit the length of TextFields sent to the HMI according to the HMI Capabilities.
+
+## Motivation
+The Mobile API defines the max length of many text fields much larger than either the SDL HMI or Generic HMI support. The true maximum length of text fields is defined by the HMI Capabilities `displayCapabilities.textFields`. For example, `AddCommand.MenuParams.tertiaryText` is defined with `maxlength="500"` in the Mobile API, but the Generic HMI's capabilities inform Core that it can handle just one twenty character wide row for `tertiaryText`. This means Core may be forwarding TextFields to the HMI that are longer than the HMI can handle.
+
+## Proposed solution
+Core will enforce a maximum length on any TextFields being sent to the HMI, based on the lengths defined for TextFields in the HMI's capabilities.
+
+### When to Truncate
+Core's ability to truncate TextFields will first be controlled by an INI option.
+```
+[HMI]
+...
++ ; Defines if Core should truncate TextFields sent to the HMI
++ ; according to the HMI Capabilities
++ TruncateTextFields = true
+```
+When this option is enabled, Core will truncate any TextFieldStruct being sent to the HMI who's `fieldName` is defined in the HMI's display capabilities.
+
+### Truncation Process
+When Core is evaluating a TextFieldStruct for truncation, it will first check if the length of the provided `TextFieldStruct.fieldText` is longer than the maximum length from the capabilities. The maximum length from the capabilities will be calculated as `TextField.width * TextField.rows`. If the provided `fieldText` is too long, the string will be cut off at the maximum length.
```

Comment:
```suggestion
When Core is evaluating a `TextFieldStruct` for truncation, it will first check if the length of the provided `TextFieldStruct.fieldText` is longer than the maximum length from the capabilities. The maximum length from the capabilities will be calculated as `TextField.width * TextField.rows`. If the provided `fieldText` is too long, the string will be cut off at the maximum length.
```